### PR TITLE
boost: Bump to version 1.76.0

### PR DESCRIPTION
**Pull request by ClaymorePT**
**Maintainer:** @ClaymorePT
**Compile tested:** BCM2711 Boards - Raspberry Pi 4 (64bits)
**Run tested:** N/A

**Description:**
This commit updates boost to version 1.76.0

There are no new libraries in this version

More info about Boost 1.76.0 can be found at the usual place [[1]].

Note: This package update includes a fix merged to Boost.Fiber in [[2]]
which did not make into this version but it will be present in the next
one. For now, the patch is needed, but it will be removed in version
1.77.0

[1]: https://www.boost.org/users/history/version_1_76_0.html
[2]: https://github.com/boostorg/fiber/pull/276

Signed-off-by: Carlos Miguel Ferreira <carlosmf.pt@gmail.com>

### CODE REVIEW

**Comment by ClaymorePT** [comment]
@neheb This PR contains the patch that you added to Fiber but didn't got into 1.76

**Comment by neheb** [comment]
Run make package/boost/refresh

File: libs/boost/Makefile [hunk 1]
```diff
@@ -11,13 +11,13 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=boost
-PKG_VERSION:=1.75.0
-PKG_SOURCE_VERSION:=1_75_0
-PKG_RELEASE:=3
+PKG_VERSION:=1.76.0
+PKG_SOURCE_VERSION:=1_76_0
+PKG_RELEASE:=1
```
Comment (neheb) [inline libs/boost/Makefile:16]: This can be replaced with AUTORELEASE to avoid changing it.

Comment (ClaymorePT) [reply libs/boost/Makefile]: @neheb Fixed in 3dceca8

Comment (brvphoenix) [reply libs/boost/Makefile]: The commit title may need to be modified.
https://github.com/openwrt/packages/issues/14537

Comment (ClaymorePT) [reply libs/boost/Makefile]: Commit updated in 7109e2e

I have to be honest. It's not something very helpful and ends up sacrificing space in the commit title

**Review by neheb — APPROVED** [review]

**Pull request merged**